src/streaming/controllers/PlaybackController.js

```diff
@@ -155,9 +155,12 @@ function PlaybackController() {
                 }
             } else {
                 eventBus.trigger(Events.PLAYBACK_SEEK_ASKED);
-                let initialStartTime = getStreamStartTime(false);
-                if (!isDynamic && time < initialStartTime) {
-                    time = initialStartTime;
+                if (!streamController.hasPreviousStream()) {
+                    //test that the user can't seek before initial start time only for the first period (for multi period use cases, don't do it for each period).
+                    let initialStartTime = getStreamStartTime(false);
+                    if (!isDynamic && time < initialStartTime) {
+                        time = initialStartTime;
```

Comment:
Hey Nico, been looking at this one myself.   If I give a non zero start time that lands in period X,  I may want to be  able to seek to any time before that initial time after stream startup.  Is this change allowing for that.  I have noticed many issues around non zero startup,  mostly when you land outside of the first period, and ability to seek back. Mostly in 3.0 vs 2.9.3 which is what we use in prod.

Reply:
Hi @dsparacio ,

you're right, there is still an issue with this PR. :-(

Nico